tests/connection/connection.cpp

```diff
@@ -1348,3 +1350,65 @@ TEST(Connection, mappingId)
                   model->component("c1")->variable("v1"),
                   model->component("c3")->variable("v3")));
 }
+
+TEST(Connection, repeatedConnection)
+{
+    const std::string in =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"connection\">\n"
+        "  <component name=\"component1\">\n"
+        "    <variable name=\"variable1\" units=\"dimensionless\"/>\n"
+        "  </component>\n"
+        "  <component name=\"component2\">\n"
+        "    <variable name=\"variable2\" units=\"dimensionless\"/>\n"
+        "  </component>\n"
+        "  <connection component_1=\"component1\" component_2=\"component2\">\n"
+        "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
+        "  </connection>\n"
+        "  <connection component_1=\"component1\" component_2=\"component2\">\n"
+        "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
+        "  </connection>\n"
+        "</model>\n";
+
+    libcellml::ParserPtr p = libcellml::Parser::create();
+    libcellml::ModelPtr m = p->parseModel(in);
+
+    const std::string expectedIssue = "Connection in model 'connection' between 'component1' and 'component2' is not unique.";
+
+    EXPECT_EQ(size_t(1), p->errorCount());
+    EXPECT_EQ(expectedIssue, p->error(0)->description());
+}
+
+TEST(Connection, repeatedMapVariables)
+{
+    const std::string in =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"connection\">\n"
+        "  <component name=\"component1\">\n"
+        "    <variable name=\"variable1\" units=\"dimensionless\"/>\n"
+        "    <variable name=\"variable6\" units=\"dimensionless\"/>\n"
+        "  </component>\n"
+        "  <component name=\"component2\">\n"
+        "    <variable name=\"variable2\" units=\"dimensionless\"/>\n"
+        "    <variable name=\"variable3\" units=\"dimensionless\"/>\n"
+        "    <variable name=\"variable5\" units=\"dimensionless\"/>\n"
+        "  </component>\n"
+        "  <connection component_1=\"component1\" component_2=\"component2\">\n"
+        "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
+        "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
```

Comment:
Might want to replace with:
```c++
"    <map_variables variable_1=\"variable2\" variable_2=\"variable1\"/>\n"
```
to confirm that we can detect non-unique connections, no matter whether which attribute is used.

Reply:
Can't because the model doesn't define the appropriate variables in the appropriate components.

Reply:
What are you saying here exactly? Are you saying that if you were to implement my suggestion then libCellML wouldn't detect that this (reversed) connection already exists?

Reply:
I think what you're suggesting @agarny is that rather than testing this by duplicating the `map_variables` this could be a duplicate connection where `_1` and `_2` are flipped for both the components and the variables.

Reply:
Looking at my original comment, I can now see the confusion. My bad! So, yes, you correctly guessed my intention @nickerso. Thus, rather than having:
```c++
        "  <connection component_1=\"component1\" component_2=\"component2\">\n"
        "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
        "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
        "    <map_variables variable_1=\"variable6\" variable_2=\"variable3\"/>\n"
        "    <map_variables variable_1=\"variable6\" variable_2=\"variable5\"/>\n"
        "    <map_variables variable_1=\"variable6\" variable_2=\"variable3\"/>\n"
        "  </connection>\n"
```
I would have:
```c++
        "  <connection component_1=\"component1\" component_2=\"component2\">\n"
        "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
        "    <map_variables variable_1=\"variable6\" variable_2=\"variable3\"/>\n"
        "    <map_variables variable_1=\"variable6\" variable_2=\"variable5\"/>\n"
        "    <map_variables variable_1=\"variable6\" variable_2=\"variable3\"/>\n"
        "  </connection>\n"
        "  <connection component_1=\"component2\" component_2=\"component1\">\n"
        "    <map_variables variable_1=\"variable2\" variable_2=\"variable1\"/>\n"
        "  </connection>\n"
```

Reply:
This is a different problem, here the connection between the two components is not unique.

Reply:
Fair enough. Then, to do something like that in `TEST(Connection, repeatedConnection)`.

Reply:
Which of course we already have a test for.

Reply:
I clearly can't see where that test is. Would you mind telling me where that test is?

Reply:
https://github.com/hsorby/libcellml/blob/9deffb34ed15cf65a72858c9f558bd22288fb982/tests/connection/connection.cpp#L1354

Reply:
Right I'm with you now (I think), is this what you are after: [ef6ab6f](https://github.com/cellml/libcellml/pull/1225/commits/ef6ab6fbb75f6cb433892a08443e9948f1d5d464)

Reply:
Yup, you are with me now. :)